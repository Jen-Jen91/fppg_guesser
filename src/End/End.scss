.end-container {
  display: flex;
  flex-direction: column;
  align-items: center;

  height: 100vh;

  background-color: #0090ff;
}

.end-title {
  background-color: #1f375b;
  width: 100vw;
  height: 80px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  color: #ffffff;
  margin: 50px 0 0 0;
}

.end-subtitle {
  @extend .end-title;
  margin: 0 0 100px 0;
  height: 90px;
}

.replay-button {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  border-radius: 20px;
  box-shadow: 0 5px #aaadb3;
  min-width: 100px;
  width: 10%;
  height: 30px;
  padding: 5px 0;

  cursor: pointer;
  background-color: #ffffff;

  &:hover {
    opacity: 0.9;
  }

  &:active {
    transform: translateY(4px);
    box-shadow: 0 2px #83858a;
  }
}

.replay-text {
  color: #1f375b;
  margin: 0;
  font-size: 1.1rem;
  font-weight: bold;
}
