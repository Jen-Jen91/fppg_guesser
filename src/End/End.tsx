import React from "react";
import { RouteComponentProps, withRouter } from "react-router";

export interface ExtendedProps extends RouteComponentProps {}

function End(props: ExtendedProps) {
  return (
    <main>
      <h1>Congratulations - you won!</h1>
      <button onClick={() => props.history.push("/")} className="replay-button">
        Replay
      </button>
    </main>
  );
}

export default withRouter(End);
