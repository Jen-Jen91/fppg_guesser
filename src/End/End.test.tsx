import React from "react";
import End from "./End";
import { shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";

test("End snapshot test", () => {
  const wrapper = shallow(
    <MemoryRouter>
      <End />
    </MemoryRouter>
  );
  expect(wrapper.render()).toMatchSnapshot();
});
