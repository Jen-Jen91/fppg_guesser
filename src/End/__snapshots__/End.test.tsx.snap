// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End snapshot test 1`] = `
<main>
  <h1>
    Congratulations - you won!
  </h1>
  <button
    class="replay-button"
  >
    Replay
  </button>
</main>
`;
